app/models/tenant.rb

```diff
@@ -155,6 +155,18 @@ def login_logo?
     !!login_logo_file_name
   end
 
+  def visible_domains
+    MiqAeDomain.where(:tenant_id => ancestor_ids.append(id)).order('priority DESC')
+  end
+
+  def enabled_domains
+    MiqAeDomain.where(:tenant_id => ancestor_ids.append(id), :enabled => true).order('priority DESC')
+  end
+
+  def editable_domains
+    MiqAeDomain.where(:tenant_id => id, :system => false).order('priority DESC')
```

Comment:
@mkanoor `editable_domains` can use the relationship already defined: `ae_domains.where(:system => false).order('priority DESC')`